dp/cloud/python/magma/configuration_controller/response_processor/strategies/response_processing.py

```diff
@@ -63,11 +63,11 @@ def process_registration_response(obj: ResponseDBProcessor, response: DBResponse
 def _find_cbsd_from_request(session: Session, payload: Dict) -> DBCbsd:
     if "cbsdSerialNumber" in payload:
         return session.query(DBCbsd).filter(
-            DBCbsd.cbsd_serial_number == payload["cbsdSerialNumber"],
+            DBCbsd.cbsd_serial_number == payload.get("cbsdSerialNumber"),
         ).scalar()
     if "cbsdId" in payload:
         return session.query(DBCbsd).filter(
-            DBCbsd.cbsd_id == payload["cbsdId"],
+            DBCbsd.cbsd_id == payload.get("cbsdId"),
```

Comment:
If there is no `cbsdId` in payload this filter will match **all** unregistered cbsd? Is that expected? I'm not sure, since we are returning scalar as if we expected only one result.

Reply:
Before constructing the query and building the cbsd filter, we are checking if the JSON payload contains the keys.
I can add failsafes for cases where the request is inavlid (does not contain neither of the keys)

Reply:
Added warning log and made sure that callers are checking if this function returns a db entity.